### ISSUE TITLE
Change Request: Replace `any` return type for `calculateConfigForFile`

### ISSUE THREAD

**Issue by vladshcherbin** [eslint/eslint#21229]
### ESLint version

v10.8.1

### What problem do you want to solve?

[Current types file](https://github.com/eslint/eslint/blob/v10.8.1/lib/types/index.d.ts#L1185) has `calculateConfigForFile` type as:

```ts
calculateConfigForFile(filePath: string): Promise<any>;
```

Would be great to have a nice type instead of `any`. [Documentation says](https://eslint.org/docs/latest/integrate/nodejs-api#return-value-3) return type is `Promise<Object>` which is different from the types file.

### What do you think is the correct solution?

I'd love `calculateConfigForFile` to have return type of `Promise<Config>` or some other appropriate type.

### Participation

- [ ] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I'm using it like this:

```ts
const { rules } = await eslint.calculateConfigForFile('test.tsx')
```

and see this lint error:

<img width="787" height="143" alt="Image" src="https://github.com/user-attachments/assets/730010cf-4a77-4ea2-af4b-a7dc79c7a6b4" />

**Comment by Pixel998** [comment]
Instead of `Promise<any>` / `Promise<Object>`, we could add an `ESLint.CalculatedConfig` type and use:
```ts
calculateConfigForFile(
    filePath: string,
): Promise<ESLint.CalculatedConfig | undefined>;
```
@eslint/eslint-team thoughts?

**Comment by snitin315** [comment]
Sounds good to me. Marking as accepted.
